gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/ServiceAuthenticationServiceImpl.java

```diff
@@ -60,24 +62,34 @@
  * also in loadbalancer
  */
 @Service
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class ServiceAuthenticationServiceImpl implements ServiceAuthenticationService {
 
     public static final String AUTHENTICATION_COMMAND_KEY = "zoweAuthenticationCommand";
 
     private static final String CACHE_BY_SERVICE_ID = "serviceAuthenticationByServiceId";
-    private static final String CACHE_BY_AUTHENTICATION = "serviceAuthenticationByAuthentication";
+    static final String CACHE_BY_AUTHENTICATION = "serviceAuthenticationByAuthentication";
 
     private final LoadBalancerAuthenticationCommand loadBalancerCommand = new LoadBalancerAuthenticationCommand();
     private final LoadBalancerAuthentication loadBalancerAuthentication = new LoadBalancerAuthentication();
 
+    private final ApplicationContext applicationContext;
     private final EurekaClient discoveryClient;
     private final EurekaMetadataParser eurekaMetadataParser;
     private final AuthenticationSchemeFactory authenticationSchemeFactory;
     private final AuthSourceService authSourceService;
     private final CacheManager cacheManager;
     private final CacheUtils cacheUtils;
 
+    // to force calling inside methods with aspects - ie. ehCache aspect
+    private ServiceAuthenticationService meAsProxy;
+
+    @PostConstruct
+    public void afterPropertiesSet() {
+        meAsProxy = applicationContext.getBean(ServiceAuthenticationService.class);
```

Comment:
maybe rather ask specifically for ServiceAuthenticationServiceImpl ? in case of different implementations in the future

Reply:
It is not possible because this bean is using aspects. In the case of a different implementation, there will still be just one instance at a time.

Reply:
Return the bean instance that uniquely matches the given object type, if any. 
Since type is ServiceAuthenticationService.class, I don't think you can be so sure.

Reply:
It is really not working: org.springframework.beans.factory.BeanNotOfRequiredTypeException: Bean named 'getServiceAuthenticationService' is expected to be of type 'org.zowe.apiml.gateway.security.service.ServiceAuthenticationServiceImpl' but was actually of type 'com.sun.proxy.$Proxy345'